support-metrics-client/src/main/java/io/confluent/support/metrics/collectors/BasicCollector.java

```diff
@@ -53,14 +51,20 @@ public BasicCollector(
    */
   @Override
   public GenericContainer collectMetrics() {
+    String kafkaVersion = AppInfoParser.getVersion();
     SupportKafkaMetricsBasic metricsRecord = new SupportKafkaMetricsBasic();
     metricsRecord.setTimestamp(time.nowInUnixTime());
-    metricsRecord.setKafkaVersion(AppInfoParser.getVersion());
-    metricsRecord.setConfluentPlatformVersion(Version.getVersion());
+    metricsRecord.setKafkaVersion(kafkaVersion);
+    metricsRecord.setConfluentPlatformVersion(cpVersion(kafkaVersion));
     metricsRecord.setCollectorState(this.getRuntimeState().stateId());
     metricsRecord.setBrokerProcessUUID(uuid.toString());
     metricsRecord.setClusterId(server.clusterId());
     return metricsRecord;
   }
 
+  // visible for testing
+  public static String cpVersion(String kafkaVersion) {
```

Comment:
I think we should make this a public utility so that other components can use it: they would need to do the same replace.

Reply:
As discussed offline, other components should use their own `Version` class instead of using this utility method.